docs/feature/dualstack/Cilium-dualstack.md

```diff
@@ -0,0 +1,23 @@
+# Cilium Azure-ipam Dualstack Overlay Proposal 
+
+## Purpose 
+
+This document proposes changes to the azure-ipam contract so that we can work with dualstack clusters between Cilium and Azure CNS. 
+
+## Overview 
+
+With the current CNS solution for dualstack we are setting the nnc to pass multiple NCs with CIDR to be unrolled and added to a pool that contains both v6 and v4 ips. Since we are planning to use Cilium CNI for the linux dualstack solution we need to make changes to azure-ipam to allow for a dualstack cluster to get multiple IPs. Currently azure-Ipam handles the [`CmdAdd`](https://github.com/Azure/azure-container-networking/blob/master/azure-ipam/ipam.go) command from Cilium CNI and sends an IP request to the CNS. The CNS then in overlay then looks through the pool and picks the first IP found that isn’t used. For dualstack we need to change two things: 1) Azure-ipam needs to be able to assign multiple IPs and 2) CNS needs to be able to determine what type of IP it is returning so that we can have one IPv4 and one IPv6. 
```

Comment:
format this list:

>... things:
>1) Azure-ipam needs to be able to assign multiple IPs and 
>2) CNS needs to be able to determine what type of IP it is returning so that we can have one IPv4 and one IPv6.

Reply:
added formatting